buoy.php

```diff
@@ -107,13 +107,24 @@ public static function registerL10n () {
      * @return void
      */
     public static function initialize () {
-        // Make sure the WP REST API plugin is installed.
-        // We can remove this if block once that plugin is merged
-        // into WP Core.
-        require_once ABSPATH.'wp-admin/includes/plugin.php';
-        if (is_plugin_inactive('rest-api/plugin.php')) {
-            if (is_wp_error(activate_plugin('rest-api/plugin.php'))) {
-                self::install_plugin_dependency('rest-api');
+        // WordPress versions prior to 4.7 did not fully merge the
+        // WP REST API, on which we (partially) rely. In version 4.7,
+        // however, the REST API's important parts for our use case,
+        // the so-called "Content API" did indeed land in core. This
+        // means we no longer need to force-install it ourselves.
+        //
+        // See https://codex.wordpress.org/Version_4.7 highlights.
+        //
+        // For now, make sure the WP REST API plugin is installed and
+        // activated for users still running older WordPress versions.
+        // We can remove this check once enough Buoys update to v4.7.
+        global $wp_version;
+        if (version_compare('4.7', $wp_version) === 1) {
```

Comment:
Now that I'm thinking about this it might make more sense for us to detect the presence or absense of the WP REST API capabilities we need directly, rather than rely on the WordPress version to do this. I don't know exactly how to do that in WordPress land though I'm sure I can figure it out.

At this point it might be moot since the version comparison is likely to be reliable across installs given we're not in a single-vendor context. Shruggie.gif